CI failures due to schema-salad

In cases like https://ucsc-ci.com/databiosphere/toil/-/jobs/19601 I was seeing CI tests fail because `schema-salad` was deciding that CWL versions like `v1.2` really ought to be filenames:

```
_________________ CWLWorkflowTest.test_download_directory_file _________________
Traceback (most recent call last):
  File "/builds/databiosphere/toil/src/toil/test/cwl/cwlTest.py", line 374, in test_download_directory_file
    self.download_directory("download_directory_file.json", self._tester)
  File "/builds/databiosphere/toil/src/toil/test/cwl/cwlTest.py", line 281, in download_directory
    tester_fn(
  File "/builds/databiosphere/toil/src/toil/test/cwl/cwlTest.py", line 228, in _tester
    cwltoil.main(main_args, stdout=st)
  File "/builds/databiosphere/toil/src/toil/cwl/cwltoil.py", line 3499, in main
    loading_context, uri = cwltool.load_tool.resolve_and_validate_document(
  File "/builds/databiosphere/toil/venv/lib/python3.9/site-packages/cwltool/load_tool.py", line 384, in resolve_and_validate_document
    document_loader.resolve_all(workflowobj, fileuri)
  File "schema_salad/ref_resolver.py", line 965, in resolve_all
  File "schema_salad/ref_resolver.py", line 1197, in validate_links
  File "schema_salad/ref_resolver.py", line 1124, in validate_links
  File "schema_salad/ref_resolver.py", line 1062, in validate_link
schema_salad.exceptions.ValidationException: src/toil/test/cwl/download_directory.cwl:1:1: Field `cwlVersion` contains undefined reference to
                                              `file:///builds/databiosphere/toil/src/toil/test/cwl/v1.2`
```

I was having trouble reproducing this on my machine, and I tracked it down to being related to the `schema-salad` version. Using version `8.3.20220518163624` or later causes the bug to appear, while using version `8.2.20220204150214` or earlier works fine.

I'm going to see if I can pin us to the older, working versions.

@mr-c Do you have any idea how we could track down what exactly the problem between us and `schema-salad` is here? Have we just been using it or `cwltool` wrong in a way that could cause it to try and validate the CWL version fields as if they were supposed to be paths to files that exist?

┆Issue is synchronized with this [Jira Story](https://ucsc-cgl.atlassian.net/browse/TOIL-1198)
┆friendlyId: TOIL-1198

--- a/requirements-cwl.txt
+++ b/requirements-cwl.txt
@@ -1,4 +1,5 @@
 cwltool==3.1.20220628170238
+schema-salad>=8,<8.3
 galaxy-tool-util
 ruamel.yaml>=0.15,<=0.17.21
 ruamel.yaml.clib>=0.2.6

